internal/pkg/peer/blocksprovider/bft_deliverer.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package blocksprovider
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/internal/pkg/identity"
+	"github.com/hyperledger/fabric/internal/pkg/peer/orderers"
+	"github.com/pkg/errors"
+)
+
+// BFTDeliverer TODO this is a skeleton
+type BFTDeliverer struct { // TODO
```

Comment:
why are we coupling gossip stuff with block delivery? 

Let's create a self contained component that only fetches blocks and is not aware of what is gossip, yielding leadership, etc.

Reply:
Done